sklearn/calibration.py

```diff
@@ -25,6 +27,337 @@
+                not hasattr(self.base_estimator, 'predict_proba'):
+            raise TypeError('The base_estimator needs to implement either a '
+                            'decision_function or a predict_proba method')
+
+        if self.method not in ['roc', 'f_beta', 'max_tpr', 'max_tnr']:
+            raise ValueError('method can either be "roc" or "max_tpr" or '
+                             '"max_tnr. Got %s instead' % self.method)
+
+        if self.scoring not in [None, 'decision_function', 'predict_proba']:
+            raise ValueError('scoring param can either be "decision_function" '
+                             'or "predict_proba" or None. Got %s instead' %
+                             self.scoring)
+
+        if self.method == 'max_tpr' or self.method == 'max_tnr':
+            if not self.threshold or not \
+                    isinstance(self.threshold, (int, float)) \
+                    or not self.threshold >= 0 or not self.threshold <= 1:
+                raise ValueError('parameter threshold must be a number in'
+                                 '[0, 1]. Got %s instead' %
+                                 repr(self.threshold))
+
+        if self.method == 'f_beta':
+            if not self.beta or not isinstance(self.beta, (int, float)) \
+                    or not self.beta >= 0 or not self.beta <= 1:
+                raise ValueError('parameter beta must be a number in [0, 1]. '
+                                 'Got %s instead' % repr(self.beta))
+
+        X, y = check_X_y(X, y)
+
+        y_type = type_of_target(y)
+        if y_type != 'binary':
+            raise ValueError('Expected target of binary type. Got %s ' %
+                             y_type)
+
+        self.label_encoder_ = LabelEncoder().fit(y)
+
+        y = self.label_encoder_.transform(y)
+        self.pos_label = self.label_encoder_.transform([self.pos_label])[0]
+
+        if self.cv == 'prefit':
+            self.decision_threshold_ = _CutoffClassifier(
+                self.base_estimator, self.method, self.beta, self.scoring,
+                self.pos_label, self.threshold
+            ).fit(X, y).decision_threshold_
+        else:
+            cv = check_cv(self.cv, y, classifier=True)
+            decision_thresholds = []
+
+            for train, test in cv.split(X, y):
+                estimator = clone(self.base_estimator).fit(X[train], y[train])
+                decision_thresholds.append(
+                    _CutoffClassifier(estimator,
+                                      self.method,
+                                      self.beta,
+                                      self.scoring,
+                                      self.pos_label,
+                                      self.threshold).fit(
+                        X[test], y[test]
+                    ).decision_threshold_
+                )
+            self.decision_threshold_ = sum(decision_thresholds) /\
```

Comment:
please avoid \
can't you just use np.mean here?

Reply:
Should we also be storing the standard deviation as a diagnostic? I wonder how reliable the decision functions are as something that can be stably thresholded.

Reply:
I agree that since we are storing the mean of the thresholds it makes sense to also store the std as a diagnostic. In case you see that the std is too high something must have gone wrong because thresholds of different splits shouldn't be too different. Is this what you mean? 

Or are you referring to the standard deviation of the actual confidence scores? Those could also serve as a diagnostic but not always easily interpretable I guess.